### DISCUSSION TITLE
Parsing of oSM0 and multiple o options

### DISCUSSION THREAD

**Discussion by cprima** [gopro/labs#348] [Q&A] [answered]
Not all options can be stacked within one single `o`:

https://gopro.github.io/labs/control/tech/ says:

> Option commands, starting with o can stack many parameter in one command

Using the GoPro QR Code Creator at https://gopro.github.io/labs/control/custom/ gives for Spot Metering Disable Spot
`oSM0`

But it seems that it is not possible to combine `oSM0` with other camera options:

- works: https://gopro.github.io/labs/control/set/?cmd=dVq1mPRoB4V4R1D4FUW1L0SX0oSM0oG1 (…oSM0o…)
- does not work: https://gopro.github.io/labs/control/set/?cmd=dVq1mPRoB4V4R1D4FUW1L0SX0SM0G1 (…SM0…)

**Posting this here because I will forget and can find my own note in a few weeks again ;)**

**Comment by cprima** [comment]
Solution: Concatenate multiple `o` segments. ;)

**Comment by cprima** [comment]
Ah, and same with the shortcuts oSA, oSB, oSC and oSD:
Does not work with a contracted `o` but need an own one.

That is sooo cool: I can now go through all my YouTube favorites and extract all "best settings" to a string of characters!

**Comment by dnewman-gpsw** [comment, marked as answer]
The options optimization might one day go way, as the "starting with o can stack many parameter in one command" was an old idea, but the commands have since increased in complexity.  oSM and oSA are pretty rare corner-cases now, but I sure there will be more in time.

**Reply by cprima** [reply]
> The options optimization might one day go way, as the "starting with o can stack many parameter in one command" was an old idea, but the commands have since increased in complexity. oSM and oSA are pretty rare corner-cases now, but I sure there will be more in time.

Thanks, good to know! (I am increasing the compllexity in my spreadsheet-based link-generator to https://gopro.github.io/labs/control/set/?cmd=  and will take this into account -- because I typically come back to such homegrown helpers after months of inactivity)

I also noticed that Nightlapse Photo and Video modes' HyperSmooth EIS seem to share the same command e2, if I am not mistaken.

**Reply by dnewman-gpsw** [reply]
Yes, e2 is context dependent (currently.)
